examples_tests/13.BRDF_Explorer/createDerivMapFromBumpMap.cpp

```diff
@@ -0,0 +1,56 @@
+#include <irrlicht.h>
+#include "createComputeShader.h"
+#include "../source/Irrlicht/COpenGL2DTexture.h"
+#include "../source/Irrlicht/COpenGLDriver.h"
+#include "CComputeShaderManager.h"
+
+using namespace irr;
+
+video::IVirtualTexture* createDerivMapFromBumpMap(video::IVirtualTexture* _bumpMap, IrrlichtDevice* _device, float _heightFactor)
+{
+    asset::IAssetManager& assetMgr = _device->getAssetManager();
+    asset::IAssetLoader::SAssetLoadParams lparams;
+
+    video::IVideoDriver* driver = _device->getVideoDriver();
+
+    const uint32_t* derivMap_sz = _bumpMap->getSize();
+    video::ITexture* derivMap = driver->createGPUTexture(video::ITexture::ETT_2D, derivMap_sz, std::log2(std::max(derivMap_sz[0], derivMap_sz[1])), asset::EF_R8G8_SNORM);
+
+    {
+        video::STextureSamplingParams params;
+        params.UseMipmaps = 0;
+        params.MaxFilter = params.MinFilter = video::ETFT_LINEAR_NO_MIP;
+        params.TextureWrapU = params.TextureWrapV = video::ETC_CLAMP_TO_EDGE;
+        const_cast<video::COpenGLDriver::SAuxContext*>(reinterpret_cast<video::COpenGLDriver*>(driver)->getThreadContext())->setActiveTexture(7, _bumpMap, params);
+    }
+
+    GLuint deriv_map_gen_cs = CComputeShaderManager::getShader("../deriv_map_gen.comp");
```

Comment:
if this is going to be in engine core, then you can build-in the shader

Reply:
@Crisspl ?

Reply:
what do you mean by build-in? If it's gonna be in engine core then i think it must be in its own class so that shader program can be deleted in the class's destructor

Reply:
resolved